src/App.js

```diff
@@ -29,37 +29,94 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
-
+  // const [winner, setWinner] = useState('');
+  const [currentPlayer, setCurrentPlayer] = useState(Player1);
   // Wave 2
   // You will need to create a method to change the square
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
 
+  const onClickSquare = (id) => {
+    console.log(id);
+    const newSquares = [...squares];
```

Comment:
Great job in making a copy of the squares data! Here's a reminder on why we need to make the copy:

To trigger the `update` stage for this component, React needs to notice a change in state data. We could mutate state directly, however, this doesn't scale well and doesn't follow React best practices. 

Instead, we should pass in a copy of our state data to the setState function `setSquares()` which will have React compare the object references between the copy and the old state data. If the object references are different, React will then trigger the `update` stage.